### ISSUE TITLE
Zero Subtotal Checkout order fails

### ISSUE THREAD

**Issue by mischabraam** [squeezely/magento2-plugin#24]
When trying to place a zero subtotal order the visitor is redirected back to the cart page and the order is not placed. A zero subtotal order can happen when (obviously) trying to buy a free product or by applying a 100% discount. 

After debugging I've found the following thrown exception.
```
/vendor/squeezely/magento2-plugin/Plugin/Invoice.php:191
DateTime::createFromFormat() expects parameter 2 to be string, null given 
```

When I disable the following piece of code in class `vendor/squeezely/magento2-plugin/Plugin/Invoice.php` placing a zero subtotal order works again. (See commented lines)
```
    public function afterPay(
        InvoiceModel $subject,
        InvoiceModel $result
    ) {
        if ($this->backendEventsRepository->isEnabled()
            && in_array(
                RequestRepository::PURCHASE_EVENT_NAME,
                $this->backendEventsRepository->getEnabledEvents()
            )
        ) {
            $this->logRepository->addDebugLog('Purchase event', __('Start'));
            try {
                $order = $subject->getOrder();
                $this->logRepository->addDebugLog(
                    'Purchase event',
                    'Order id: ' . $order->getIncrementId()
                );
                if ($subject->getState() === Order\Invoice::STATE_PAID) {
//                    $this->requestRepository->sendPurchases(
//                        $this->transformOrderData($order)
//                    );
                }
            } catch (\Exception $exception) {
                $this->logRepository->addErrorLog('afterPay plugin', $exception->getMessage());
            }
        }
        $this->logRepository->addDebugLog('Purchase event', __('Finish'));
        return $result;
    }
```

**Comment by Marvin-Magmodules** [comment]
Hi @mischabraam, sorry for the late reaction. The first issue was fixed in this commit; https://github.com/squeezely/magento2-plugin/commit/22d0f2891be1db1e7504ab56d693025b10d7b540 and released into version 2.4.3. What version are you running at the moment? 

We are not able to reproduce this issue anymore on the latest version.

**Comment by mischabraam** [comment]
Squeezely module running at the time was 2.4.1. I've tested with version 2.7.0 in which the issue seems to be resolved. Thanks.
